### ISSUE TITLE
Steam version: godot runs application, and steam eats memory each time it runned + UI Overlay problem

### ISSUE THREAD

**Issue by PixelSymbols** [godotengine/godot#79732]
### Godot version

4.1.stable

### System information

Windows 11, steam version

## Steam overlay works inside Godot
![image](https://github.com/godotengine/godot/assets/81364140/90a9cd4a-8741-4f7f-969c-ca1365a5a4a5)
![image](https://github.com/godotengine/godot/assets/81364140/3ee625cc-990e-490c-9923-d7999ca166c1)
even tho its disabled

and this maybe what's causing it to eat heavily memory.
this is my memory usage right now:
![image](https://github.com/godotengine/godot/assets/81364140/0bae72eb-4f1f-44b0-a80d-33a8f4adbbd2)
if ill run the project 15 times (by F5)
![image](https://github.com/godotengine/godot/assets/81364140/d8c673c3-a6a0-4fed-ae23-2e212791629e)
ram will go up
if ill continue doing so 30-40 times more:
![image](https://github.com/godotengine/godot/assets/81364140/b6303810-3857-4abd-a1dd-1590a320f901)
it will now be +5 gb. And the most weird thing, that its not the godot issue itself, but a steam
![image](https://github.com/godotengine/godot/assets/81364140/7297f702-dea2-4a1b-91e9-2ed956a15a56)
for whatever reason after you compile and run the project in godot, steam each time eats memory.

So i guess both these issues are connected. That steam does not clean up memory after godot game was closed
![image](https://github.com/godotengine/godot/assets/81364140/24959742-016e-42d5-860c-9be81a057d31)

So there are 2 bugs, one bug is from steam, about memory eating. But i guess the reason is that godot does not send to steam info that game inside godot was closed, which is why steam keeps ui overlay in memory.
Also maybe godot fault or maybe steam fault that disabling (disabled by default) steam's ui overlay for godot does not work, it still works both inside the game engine and the game 🤔

also my current steam settings (maybe its important):
![image](https://github.com/godotengine/godot/assets/81364140/695f6b48-5751-4c9e-8564-803f381d53c7)

### Steps to reproduce

Download Godot in steam
run project
run project multiple times

### Minimal reproduction project

Download Godot in steam
run project
run project multiple times

**Comment by PixelSymbols** [comment]
> 

no, i can't, in project manager overlay does not works, only in the editor and in the game

**Comment by PixelSymbols** [comment]
> This is likely because going from the project manager to editor starts a subprocess, and Steam isn't aware of this. (This is what causes the time tracking issues.)

So, is it possible to fix right now and will it be fixed in the future, or its unlikely?

**Comment by Calinou** [comment]
> So, is it possible to fix right now and will it be fixed in the future, or its unlikely?

https://github.com/godotengine/godot/pull/79126 could help with this, but it would add complexity to the buildsystem as described in its comments.

**Comment by Zireael07** [comment]
https://github.com/godotengine/godot/issues/79639 is likely related to this one too (the latest comment mentions steam overlay)
